gnomad_hail/resources/resource_utils.py

```diff
@@ -9,114 +9,221 @@
+        :param kwargs: Any other parameters to be passed to hl.Table.write
+        :return: Nothing
+        """
+        self.import_func(**self.import_sources).write(
+            self.path,
+            overwrite=overwrite,
+            **kwargs
+        )
 
 
 class MatrixTableResource(BaseResource):
     """
     A Hail MatrixTable resource
+
+    :param path: The MatrixTable path (typically ending in .mt)
+    :param import_sources: Any sources that are required for the import and need to be kept track of and/or passed to the import_func (e.g. .vcf path for an imported VCF)
+    :param import_func: A function used to import the MatrixTable. `import_func` will be passed the `import_sources` dictionary as kwargs.
     """
 
-    def __init__(self, path: str, import_sources: Optional[Dict[str, Any]] = None, ):
+    def __init__(
+            self,
+            path: Optional[str] = None,
+            import_sources: Optional[Dict[str, Any]] = None,
+            import_func: Optional[Callable[..., hl.MatrixTable]] = None
+    ):
         super().__init__(
             path=path,
             import_sources=import_sources,
-            expected_file_extension='.mt'
+            import_func=import_func,
+            expected_file_extensions=['.mt']
         )
 
-    def mt(self) -> hl.MatrixTable:
+    def mt(self, force_import: bool = False) -> hl.MatrixTable:
         """
         Read and return the Hail MatrixTable resource
 
         :return: Hail MatrixTable resource
         """
-        return hl.read_matrix_table(self.path)
+        if self.path is None or force_import:
+            return self.import_func(**self.import_sources)
+        else:
+            return hl.read_matrix_table(self.path)
+
+    def import_resource(self, overwrite: bool = False, **kwargs):
+        """
+        Imports the MatrixTable resource using its import_func and writes it in its path.
+
+        :param overwrite: If set, existing file(s) will be overwritten
+        :param kwargs: Any other parameters to be passed to hl.MatrixTable.write
+        :return: Nothing
+        """
+        self.import_func(**self.import_sources).write(self.path, overwrite=overwrite, **kwargs)
 
 
 class PedigreeResource(BaseResource):
     """
     A pedigree resource
+
+    :param path: The Pedigree path (typically ending in .fam or .ped)
+    :param import_sources: Any sources that are required for the import and need to be kept track of and/or passed to the import_func (e.g. .vcf path for an imported VCF)
+    :param import_func: A function used to import the MatrixTable. `import_func` will be passed the `import_sources` dictionary as kwargs.
+    :param quant_pheno: If ``True``, phenotype is interpreted as quantitative.
+    :param delimiter: Field delimiter regex.
+    :param missing: The string used to denote missing values. For case-control, 0, -9, and non-numeric are also treated as missing.
```

Comment:
These new options don't really seem like they're in scope of this PR. @mike-w-wilson can you take a look at these?

Reply:
I think these were overlooked before -- they are IMHO helpful to have here as they contain key information regarding the resource that should be easily accessible.

Reply:
I haven't worked with pedigrees much in hail but agree they seem a bit out of place since we do not define any of the other parameters for hail functions for other resources.